### PR TITLE
Add a pass to normalize address spaces before lowering to standard ops.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**
This takes care of handling memrefs with address space '2'.

### CODE REVIEW

File: python/compiler/aiecc/main.py [hunk 1]
```diff
@@ -694,6 +694,7 @@ async def run_flow(self):
         if(opts.unified):
           self.file_opt_with_addresses = os.path.join(self.tmpdirname, 'input_opt_with_addresses.mlir')
           await self.do_call(progress_bar.task, ['aie-opt', '--aie-localize-locks',
+                              '--aie-normalize-address-spaces',
```
Comment (stephenneuendorffer) [inline python/compiler/aiecc/main.py:697]: This is already part of aie-opt-passes?

Comment (MaheshRavishankar) [reply python/compiler/aiecc/main.py:697]: Yes, but it was suggested that this has to run before conversion to standard based on an error with handle address space 2

Comment (stephenneuendorffer) [reply python/compiler/aiecc/main.py:697]: Thanks, After your patch, I went back and actually processed what Jeff had indicated.

Comment (stephenneuendorffer) [reply python/compiler/aiecc/main.py:697]: It doesn't make sense to me to run this pass twice... so I factored it out of aie_opt_passes

**Review by fifield — APPROVED** [review]

**Comment by makslevental** [comment]
@stephenneuendorffer can I merge?

**Pull request merged**